models/pull.go

```diff
@@ -266,7 +266,7 @@ func (pr *PullRequest) Merge(doer *User, baseGitRepo *git.Repository, mergeStyle
 		if _, stderr, err = process.ExecDir(-1, tmpBasePath,
 			fmt.Sprintf("PullRequest.Merge (git merge): %s", tmpBasePath),
 			"git", "commit", fmt.Sprintf("--author='%s <%s>'", sig.Name, sig.Email),
-			"-m", fmt.Sprintf("Merge branch '%s' of %s/%s into %s", pr.HeadBranch, pr.HeadUserName, pr.HeadRepo.Name, pr.BaseBranch)); err != nil {
+			"-m", strings.TrimSpace(fmt.Sprintf("Merge branch '%s' of %s/%s into %s\n%s", pr.HeadBranch, pr.HeadUserName, pr.HeadRepo.Name, pr.BaseBranch, CommitMessage))); err != nil {
```

Comment:
You are right, the additional description should be separated from the first line by an empty line.
Although this is not required by git, it seems to be a good practice many tools follow.

I am not so sure about a construct like

```go
args := []string{...,...}
if len(CommitMessage) > 0 {
    args = append(args, "-m", CommitMessage)
}
process.ExecDir(..., ..., "git", args...)
```

Although we already do this on several other occations, I feel it makes the code unnecessary complex, as by the documentation it is the same as concatenating with double newlines.

<quote>
-m <msg>, --message=<msg>
Use the given <msg> as the commit message. If multiple -m options are given, their values are concatenated as separate paragraphs.
</quote>

~~The TrimSpace variant also elegantly solves the case, where someone accidentally enters e.g. a simple space into the textarea~~

Okay, git seems to be smart enough to drop empty (whitespace only) paragraphs anyway, so we could actually always give it a second "-m" parameter for the description and let git drop that paragraph if it is empty. What's your take on this?

Reply:
>Okay, git seems to be smart enough to drop empty (whitespace only) paragraphs anyway, so we could actually always give it a second "-m" parameter for the description and let git drop that paragraph if it is empty. What's your take on this?

Sounds good if it actually works.

Reply:
So I just did some checks using the following code:

```bash
git init
touch a
git add a
git commit -m "msg" -m ""
git log
```

First using git version 1.7.10.4 which is the oldest one I currently have.

Result:

```
$ git log
commit f39655f54bcc19a9f7df3e5fc98e0da856aab4d3
Author: [omitted]
Date:   Mon Jun 25 13:11:56 2018 +0200

    msg
$
```

Then again using git version 2.17.1

```
$ git log
commit fe8bc37b9af5097e6c8a440fb285eb8833d103a6 (HEAD -> master)
Author: [omitted]
Date:   Mon Jun 25 13:13:21 2018 +0200

    msg
$ 
```

So that behaviour looks pretty stable.

I guess it also needs to be that way, since when you edit the commit message in your editor, you also want git to drop not only the commented explanation but also all the whitespace between your text and this comment.

```

msg

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
# ...
```

Reply:
Looks awesome!